frontend/www/js/omegaup/components/course/Statistics.vue

```diff
@@ -123,28 +169,98 @@ export default class Statistics extends Vue {
     }
     return maxPoints;
   }
-  get maxVariance() {
-    let maxVariance = 0;
-    for (const variance of this.getStatistic('variance')) {
-      if (variance > maxVariance) maxVariance = variance;
+  getMaxStat(statistic: 'variance' | 'avg_runs') {
+    let max = 0;
+    for (const stat of this.getStatistic(statistic)) {
+      if (stat > max) max = stat;
+    }
+    return max;
+  }
+  problemIndices() {
+    const indices: {
+      [assignmentAlias: string]: { [problemAlias: string]: number };
+    } = {};
+    this.problemStats.forEach((problem, index) => {
+      if (!indices.hasOwnProperty(problem.assignment_alias))
+        indices[problem.assignment_alias] = {};
+      indices[problem.assignment_alias][problem.problem_alias] = index;
+    });
+    return indices;
+  }
+  get runsPerAssignment() {
+    const problems: string[] = this.problems;
+    let runSum: number[] = new Array(problems.length).fill(0);
+    for (const stat of this.verdicts) {
+      runSum[
+        this.problemIndices()[stat.assignment_alias][stat.problem_alias]
+      ] += stat.runs;
+    }
+    return runSum;
+  }
+  verdictList(): string[] {
+    let verdicts: string[] = [];
+    for (const stat of this.verdicts) {
+      if (stat.verdict && !verdicts.includes(stat.verdict))
+        verdicts.push(stat.verdict);
+    }
+    return verdicts;
+  }
+  get verdictStats() {
+    let verdicts: string[] = this.verdictList();
+    const assignmentRuns: number[] = this.runsPerAssignment;
+    const problemCount: number = this.problems.length;
+    let series: { name: string; data: number[] }[] = [];
+    const indices = this.problemIndices;
```

Comment:
```suggestion
    const indices = this.problemIndices();
```

otherwise L226 becomes super expensive.